### ISSUE TITLE
bad output with async/await

### ISSUE THREAD

**Issue by jamestalmage** [power-assert-js/babel-plugin-espower#6]
I am using ava:

```js
// test.js
import test from 'ava';

function foo() {
	return Promise.resolve(true);
}

test('foo', async t => {
	t.plan(1);
	t.ok(await foo() === false);
});
```

output
```
  Error: Line 1: Unexpected identifier
    at Test.<anonymous> (/.../test.js:10:4)
    at step (/.../test.js:5:273)
```

I have cross-posted at https://github.com/sindresorhus/ava/issues/113, since I am not sure which library is causing the issue. Feel free to close if it turns out to be ava.

**Comment by twada** [comment]
@jamestalmage Thank you for reporting!

In fact, this is a hard problem since async/await is not in standard yet, so standard-compliant parsers doesn't support them.

For now, please avoid using await expression __INSIDE__ assertions.

Doesn't work:
```js
test('foo', async t => {
    t.plan(1);
    t.ok(await foo() === false);
});
```

Would work:
```js
test('foo', async t => {
    t.plan(1);
    const result = await foo();
    t.ok(result === false);
});
```

Sorry for inconvenience, I'll fix it ASAP.

**Comment by jamestalmage** [comment]
What parser does `power-assert` use (I am assuming babylon, at least when used as a babel plugin), and I am pretty sure babylon supports async/await.

Also, this problem still occurs with the generator/yield equivalent

```js
test('foo', function *(t) {
    t.plan(1);
    t.ok(yield foo() === false);
});
```

generators are in standard.

**Comment by twada** [comment]
> What parser does `power-assert` use (I am assuming babylon, at least when used as a babel plugin), and I am pretty sure babylon supports async/await.

babylon for the transformation side and [esprima](https://github.com/jquery/esprima) for the runtime side.
I've designed runtime side to run under old browsers such as IE8, parsers written in ES3 could only be embedded in power-assert.

I know, it's time to drop support of too old browsers...

> Also, this problem still occurs with the generator/yield equivalent
> generators are in standard.

Sorry. `YieldExpression` is not supported yet and will be supported in next version.

**Comment by jamestalmage** [comment]
Acorn publishes [ECMAScript 3 builds](https://github.com/ternjs/acorn#build-system).

According to [this eslint/espree issue](https://github.com/eslint/espree/issues/200), Acorn provides some "hooks" that made the transition from esprima fairly easy.

**Comment by jamestalmage** [comment]
> Acorn...

Which would totally be helpful information if Acorn supported async/await (it does not). :blush:

**Comment by jamestalmage** [comment]
fixed in `1.1.0`:
```

              # test/some_test.js:6
              
              assert((yield bigOrSmall(input)) === big)
                      |                |       |   |   
                      |                |       |   "big"
                      "small"          3       false   
             
              --- [string] big',
              +++ [string] yield bigOrSmall(input)
              @@ -1,3 +1,5 @@
              -big
              +small        
```
